src/sandbox.cc

```diff
@@ -72,14 +84,28 @@ static int COMPILER_SYSCALLS[] = {
   SYS_getsockopt,
   SYS_fadvise64,
   SYS_shutdown,
+#ifdef __riscv
+  SYS_ppoll,
+#else
   SYS_poll,
+#endif
   -1
 };
 
 static int MOST_SYSCALLS[] = {
```

Comment:
This is getting a little bit hard to review. I wonder if it would be better to go for separate lists (rather than all these ifdefs) long term. For now, this is good enough for me.

Reply:
The syscalls should eventually converge as RISC-V/Linux support matures; but in the meantime I agree that this is very ugly.

If you prefer, I can consolidate the #ifndef __riscv 's together, which would clean up a lot of it.  I wasn't sure if that was ideal, or keeping the syscalls in the original order.

Reply:
Let's keep it as it is now and revisit later!

Reply:
Sounds good.